### PR TITLE
Feature/lattice 2312 permissions audit

### PR DESCRIPTION
**Pull request by pckeyes**
Adds auditing to permissions updates
Removes repeated logic for getting current user Id

### CODE REVIEW

**Review by you-leee — APPROVED** [review]
LGTM

File: src/main/java/com/openlattice/datastore/data/controllers/DataController.java [hunk 1]
```diff
@@ -937,10 +937,6 @@ public long getEntitySetSize( @PathVariable( ENTITY_SET_ID ) UUID entitySetId )
         return dgm.getLinkedEntitySetBreakDown( entityKeyIdsOfEntitySets, authorizedPropertyTypesOfEntitySets );
     }
 
-    private UUID getCurrentUserId() {
```
Comment (you-leee) [inline src/main/java/com/openlattice/datastore/data/controllers/DataController.java:940]: 🎉

File: src/main/java/com/openlattice/datastore/search/controllers/SearchController.java [hunk 2]
```diff
@@ -710,10 +710,6 @@ public Void triggerOrganizationIndex( @PathVariable( ORGANIZATION_ID ) UUID orga
         return null;
     }
 
-    private UUID getCurrentUserId() {
```
Comment (you-leee) [inline src/main/java/com/openlattice/datastore/search/controllers/SearchController.java:713]: 🎉

File: src/main/java/com/openlattice/datastore/permissions/controllers/PermissionsController.java [hunk 3]
```diff
@@ -248,11 +262,33 @@ public AuthorizationManager getAuthorizationManager() {
...
-        Set<AclKey> aclKeys = acls.stream().map(acl -> new AclKey( acl.getAclKey() )).collect( Collectors.toSet() );
+        Set<AclKey> aclKeys = acls.stream().map( acl -> new AclKey( acl.getAclKey() ) ).collect( Collectors.toSet() );
         Set<AclKey> allOrgExternalDBAclKeys = securableObjectResolveTypeService
                 .getOrganizationExternalDatabaseAclKeys( aclKeys );
         return acls.stream().filter( acl -> allOrgExternalDBAclKeys.contains( acl.getAclKey() ) )
                 .collect( Collectors.toList() );
     }
+
+    private List<AuditableEvent> createAuditableEvents( List<Acl> acls, AuditEventType eventType ) {
+        return acls.stream().map( acl ->
```
Comment (makosblade) [inline src/main/java/com/openlattice/datastore/permissions/controllers/PermissionsController.java:280]: ~~I think in this case (`acls` being a concrete List and never a lazy list/streamed collection), prefer using `.map {}` over `.stream().map(`~~

Okay so nevermind b/c this is java but I think the above is true in kotlin in general

**Review by makosblade — APPROVED** [review]

**Review by dbt — APPROVED** [review]

**Pull request merged**